challengeMk2/ViewModels/SystemDetailCarouselViewModel.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using challengeMk2.Models;
+using Newtonsoft.Json;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+
+namespace challengeMk2.ViewModels
+{
+    public class SystemDetailCarouselViewModel : BaseViewModel
+    {
+        private StarSystem currentSystem;
+        public StarSystem CurrentSystem
+        {
+            get => currentSystem;
+            set
+            {
+                SetProperty<StarSystem>(ref currentSystem, value);
+            }
+        }
+
+        private StarSystem detailedSystem;
+        public StarSystem DetailedSystem
+        {
+            get => detailedSystem;
+            set
+            {
+                SetProperty<StarSystem>(ref detailedSystem, value);
+            }
+        }
+
+
+        private int currentBodyCount;
+        public int CurrentBodyCount
+        {
+            get => currentBodyCount;
+            set
+            {
+                SetProperty<int>(ref currentBodyCount, value);
+            }
+        }
+
+        private double currentDistance;
+        public double CurrentDistance
+        {
+            get => currentDistance;
+            set
+            {
+                SetProperty<double>(ref currentDistance, value);
+            }
+        }
+
+
+        public NetworkAccess CurrentConnectivity { get; set; }
+
+        public IList SystemInfos { get; set; }
+
+
+
+
+
+
+
+
+        public SystemDetailCarouselViewModel(StarSystem selectedSystem = null)
+        {
+            Title = "Star System Details";
+
+            SystemInfos = new ObservableCollection<StarSystem>();
+
+            currentSystem = selectedSystem;
+
+            //Get infos
+            UpdateSystemData();
```

Comment:
It's not a good pattern to initialize data in the constructor.
More generally, a constructor should never perform any expensive work (e.g. I/O), and is generally not expected to fail (except for basic argument validation).